api/middleware/applicationMiddleware.js

```diff
@@ -51,25 +46,13 @@ const cacheSignUpData = async (req, res, next) => {
 
   try {
     if (role === 'mentor') {
-      if (!newMentorApplication.first_name) {
-        next({ status: 400, message: 'first_name required' });
-      } else if (!newMentorApplication.last_name) {
-        next({ status: 400, message: 'last_name required' });
-      } else if (!newMentorApplication.email) {
-        next({ status: 400, message: 'email required' });
-      } else if (!newMentorApplication.country) {
-        next({ status: 400, message: 'country required' });
-      } else if (!newMentorApplication.state) {
-        next({ status: 400, message: 'state required' });
-      } else if (!newMentorApplication.city) {
-        next({ status: 400, message: 'city required' });
-      } else {
-        req.body.position = 3;
-        await insertMentorIntake(newMentorApplication);
-        next();
-      }
-    } else if (role === 'mentee') {
+      req.body.position = 3;
+      sendData(newMentorApplication, role);
+      await insertMentorIntake(newMentorApplication);
+      next();
+    } else {
       req.body.position = 4;
+      sendData(newMentorApplication, role);
       await insertMenteeIntake(newMenteeApplication);
       next();
     }
```

Comment:
We could remove this if/else block. The only difference between the two cases, that I can see, is what we attach to `req.body.position`. We could do `req.body.position = role === 'mentor' ? 3 : 4;` and save several lines of code. On a side note, should we be passing in `newMenteeApplication` on line 55 instead? If so, the if/else block is probably still necessary.

Reply:
Yeah, we have to pass in the application at that point, as well as now each block creates a req object for the new changes I implemented. If you want to go over it on a call we can.